### PR TITLE
BDS#128-Download List

### PR DESCRIPTION
**Pull request by rotwurstesser**

### CODE REVIEW

File: app/components/ch/components/DownloadList.vue [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    <button
+      class="download-item"
+      :href="url"
+      target="_blank"
+      :download="title"
+      :id="`download-item-${id}`"
+    >
+      <SvgIcon icon="Download" size="xl" class="download-item-icon" />
+      <div>
+        <p class="download-item-title">
```
Comment (sylvainaerni) [inline app/components/ch/components/DownloadList.vue:12]: We use pure old BEM nomenclature in the project, so please rename your classes like `download-item__title` in place of `download-item-title`

File: app/components/ch/components/DownloadList.vue [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+          {{ title }}
+        </p>
+        <p v-if="description" class="download-item-description">
+          {{ description }}
+        </p>
+        <p class="download-meta-info">
```
Comment (sylvainaerni) [inline app/components/ch/components/DownloadList.vue:18]: What about something more generic, like `meta-info` ? We may reuse this somewhere else ?

File: app/components/ch/components/DownloadList.vue [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+          <span class="download-item-size">
+            {{ type }}
+          </span>
+          <span class="download-item-meta-divider">|</span>
```
Comment (sylvainaerni) [inline app/components/ch/components/DownloadList.vue:22]: As this is a visual separator only, isn't it better to drop the markup and do it with CSS?

File: app/components/stories/components/DownloadList.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  >
+    {Template.bind({})}
+  </Story>
+</Canvas>
+
+# Interaction and accessibility
+
+Each DownloadList item is it's own button, they  can also be added within a list and their width is contained by the outer parent element.
+
+```html
```
Comment (sylvainaerni) [inline app/components/stories/components/DownloadList.stories.mdx:36]: It's a bit dangerous to write manually markup here. Chances are big that we forget to update it if we change something in the component. As the HTML is already in the Storybook/Canvas tab, I would suggest to remove it.

File: app/components/stories/components/DownloadList.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    <p class="download-meta-info"><span class="download-item-size">
+        PDF
+      </span> <span class="download-item-meta-divider">|</span> <span class="download-item-date">
+        13.01.2022
+      </span></p>
+  </div>
+</button>
+```
+
+During interactions, the DownloadList items are highlighted and the user can click on them to trigger the download. The DownloadList items are also accessible by tabbing through them.
```
Comment (sylvainaerni) [inline app/components/stories/components/DownloadList.stories.mdx:65]: I like your explanations there

File: app/components/stories/components/DownloadList.stories.mdx [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+```javascript
+const DownloadItem = {
+  init(target) {
+    // trigger download intead of routinf on click
```
Comment (sylvainaerni) [inline app/components/stories/components/DownloadList.stories.mdx:70]: It doesn't work on my machine. The document opens in the same window. 
And `routing` in place of `routinf`

File: app/components/stories/components/DownloadList.stories.mdx [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+export default DownloadItem
+```
+
+<a href="?path=/story/components-DownloadList--example">
+  Go to the Canvas Tab
+</a>
+
+---
+
+<a href="?id=components-DownloadList--example" target="_blank">
```
Comment (sylvainaerni) [inline app/components/stories/components/DownloadList.stories.mdx:94]: This url is broken, use `?id=components-downloadlist--example` instead

File: css/components/download-item.postcss [hunk 8]
```diff
@@ -0,0 +1,28 @@
+/*----------------------------------------*\
+  Download Item
+\*----------------------------------------*/
+
+.download-item {
+  @apply flex pl-0 pb-4 pt-4 align-top border-b border-gray-200 text-left w-full;
+  @apply hover:bg-secondary-100;
```
Comment (sylvainaerni) [inline css/components/download-item.postcss:7]: I prefer `hover:bg-secondary-50` here

File: css/components/download-item.postcss [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+}
+
+.download-item-icon {
+  @apply text-red-600 pr-6;
```
Comment (sylvainaerni) [inline css/components/download-item.postcss:11]: adapting for mobile and tablets with `pr-2 md:pr-4 xl:pr-6` would be great

File: css/components/download-item.postcss [hunk 10]
```diff
@@ -0,0 +1,28 @@
+/*----------------------------------------*\
+  Download Item
+\*----------------------------------------*/
+
+.download-item {
+  @apply flex pl-0 pb-4 pt-4 align-top border-b border-gray-200 text-left w-full;
```
Comment (sylvainaerni) [inline css/components/download-item.postcss:6]: I try to regroup logically the properties for better readability. What about
```
@apply flex w-full;
@apply pl-0 pb-4 pt-4;
@apply border-b border-gray-200;
```
`align-top`can be removed

File: app/components/ch/components/DownloadList.vue [hunk 11]
```diff
@@ -0,0 +1,73 @@
+<template>
+  <li>
+    <button
```
Comment (sylvainaerni) [inline app/components/ch/components/DownloadList.vue:3]: To be double checked, but looks like it's better to use a link with `download` attribute in place of a button element:
https://css-tricks.com/building-good-download-button/

File: app/components/ch/components/DownloadList.vue (hunk 2 above)
Comment (rotwurstesser) [reply app/components/ch/components/DownloadList.vue]: yep. done

File: app/components/ch/components/DownloadList.vue (hunk 3 above)
Comment (rotwurstesser) [reply app/components/ch/components/DownloadList.vue]: yep. done

File: css/components/download-item.postcss (hunk 8 above)
Comment (rotwurstesser) [reply css/components/download-item.postcss]: updated

File: css/components/download-item.postcss (hunk 10 above)
Comment (rotwurstesser) [reply css/components/download-item.postcss]: check and check

File: app/components/ch/components/DownloadList.vue (hunk 11 above)
Comment (rotwurstesser) [reply app/components/ch/components/DownloadList.vue]: done and done

File: css/components/download-item.postcss (hunk 9 above)
Comment (rotwurstesser) [reply css/components/download-item.postcss]: true, done

**Pull request merged**